src/utils/FocusList.ts

```diff
@@ -0,0 +1,114 @@
+export class FocusList<T> {
+    private readonly array: T[];
+    private focus: number;
+
+    constructor(array: T[] = [], focus: number = 0) {
+        this.array = array
+        this.focus = focus
+    }
+
+    length() {
+        return this.array.length;
+    }
+
+    getFocus() {
+        return this.focus
+    }
+
+    setFocus(focus: number) {
+        if(focus >= this.length())
+        this.focus = focus
+    }
+
+    getContent() {
+        return this.array;
+    }
+
+    getItemInFocus() {
+        return this.array[this.focus]
+    }
+
+    shiftFocusLeft() {
+        if (this.focus === 0) this.focus = this.length() - 1
+        else this.focus = this.focus - 1
+    }
+
+    shiftFocusRight() {
+        if (this.focus === this.length() - 1) this.focus = 0
```

Comment:
```suggestion
        if (this.focus >= this.length() - 1) this.focus = 0
```
also covers you in case the value ever gets accidentally higher than it should